app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __setattr__(self, attr: str, value: int) -> None:
+        if attr == "health" and value <= 0 and self in Animal.alive:
+            Animal.alive.remove(self)
+        super().__setattr__(attr, value)
```

Comment:
why did you override the `__setattr__` method?
it is better to create a separate method for this